ui/main/src/app/modules/free-message/free-message.component.ts

```diff
@@ -122,21 +122,17 @@ export class FreeMessageComponent implements OnDestroy {
         this.entityOptions$ = this.store.select(selectAllEntities).pipe(
             takeUntil(this.unsubscribe$),
             map((allEntities: Entity[]) => allEntities.map((entity: Entity) => {
-                    return {value: entity.id, label: entity.name};
-                })
+                return { value: entity.id, label: entity.name };
+            })
             )
         );
-        this.store.select(selectIdentifier)
-            .pipe(
-                switchMap(id => this.userService.askUserApplicationRegistered(id)),
-                withLatestFrom(this.store.select(selectProcesses))
-            )
-            .subscribe(([user, allProcesses]: [User, Process[]]) => {
-
-                this.currentUser = user;
+        this.store.select(selectProcesses).pipe(takeUntil(this.unsubscribe$)).
+            subscribe((allProcesses: Process[]) => {
+                const userWithPerimeters = this.userService.getCurrentUserWithPerimeters();
+                if (userWithPerimeters) {
+                    this.currentUser = userWithPerimeters.userData;
```

Comment:
These three line should be put out of the subscribe

Reply:
Ok

Reply:
?